pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -186,23 +183,29 @@ func NewDeploymentController(client kube.Client, clusterID cluster.ID, env *mode
 	dc.serviceAccounts.AddEventHandler(parentHandler)
 	dc.clients[gvr.ServiceAccount] = NewUntypedWrapper(dc.serviceAccounts)
 
-	dc.namespaces = kclient.New[*corev1.Namespace](client)
-	dc.namespaces.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
-		// TODO: make this more intelligent, checking if something we care about has changed
-		// requeue this namespace
-		for _, gw := range dc.gateways.List(o.GetName(), klabels.Everything()) {
-			dc.queue.AddObject(gw)
-		}
-	}))
+	dc.gateways = kclient.New[*gateway.Gateway](client)
+	dc.gateways.AddEventHandler(controllers.ObjectHandler(dc.queue.AddObject))
 
-	gateways.AddEventHandler(controllers.ObjectHandler(dc.queue.AddObject))
-	gatewayClasses.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
-		for _, g := range dc.gateways.List(metav1.NamespaceAll, klabels.Everything()) {
-			if string(g.Spec.GatewayClassName) == o.GetName() {
-				dc.queue.AddObject(g)
+	if !client.IsMultiTenant() {
+		dc.namespaces = kclient.New[*corev1.Namespace](client)
+		dc.namespaces.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
+			// TODO: make this more intelligent, checking if something we care about has changed
+			// requeue this namespace
+			for _, gw := range dc.gateways.List(o.GetName(), klabels.Everything()) {
+				dc.queue.AddObject(gw)
 			}
-		}
-	}))
+		}))
+		dc.gatewayClasses = kclient.New[*gateway.GatewayClass](client)
```

Comment:
> Do you know if it was because the gatewayclass controller in istiod requires create&update privileges for gatewayclasses?

I think it's because `GatewayClass` is cluster-scoped resource.

Reply:
Yeah, I'm trying to remember why we thought having access to cluster-scoped resources was such a bad thing. For some resources (such as GatewayClass), having read-only privileges don't seem like such a bad thing. After all, the whole purpose of Gateway API is allowing regular users to create their own gateways. And for that, they need access to GatewayClasses.